From d83402ef2455f6b43126f0120106e9549ddd1775 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 11:36:38 -0400
Subject: [ethernet] i40e: Let MDD events be handled by MDD handler

Message-id: <1409658042-22689-38-git-send-email-sassmann@redhat.com>
Patchwork-id: 90232
O-Subject: [RHEL7.1 PATCH 037/281] i40e: Let MDD events be handled by MDD handler
Bugzilla: 1091126
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Anjali Singhai Jain <anjali.singhai@intel.com>

We have a separate handler for MDD events, a generic reset is not required.

Change-ID: I77858e2d479e4e65c52aede67109464649ea0253
Signed-off-by: Anjali Singhai Jain <anjali.singhai@intel.com>
Tested-by: Kavindya Deegala <kavindya.s.deegala@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit c0c289759c815a67f176d6f8fa0e44a97f27e46d)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40e/i40e_main.c b/drivers/net/ethernet/intel/i40e/i40e_main.c
index 669715b..54e1462 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_main.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_main.c
@@ -2892,8 +2892,7 @@ static irqreturn_t i40e_intr(int irq, void *data)
     icr0_remaining);
   if ((icr0_remaining & I40E_PFINT_ICR0_PE_CRITERR_MASK) ||
       (icr0_remaining & I40E_PFINT_ICR0_PCI_EXCEPTION_MASK) ||
-      (icr0_remaining & I40E_PFINT_ICR0_ECC_ERR_MASK) ||
-      (icr0_remaining & I40E_PFINT_ICR0_MAL_DETECT_MASK)) {
+      (icr0_remaining & I40E_PFINT_ICR0_ECC_ERR_MASK)) {
    dev_info(&pf->pdev->dev, "device will be reset\n");
    set_bit(__I40E_PF_RESET_REQUESTED, &pf->state);
    i40e_service_event_schedule(pf);
-- 
1.7.1